docs/api.md

```diff
@@ -2344,6 +2353,175 @@ const puppeteer = require('puppeteer');
+
+In this example, Legacy Mode prevails and the request is aborted immediately because at least one handler omits `priority` when resolving the intercept:
+
+```ts
+// Final outcome: immediate abort()
+page.setRequestInterception(true);
+page.on('request', (request) => {
+  request.abort('failed'); // Legacy behavior: would execute immediately
+});
+page.on('request', (request) => {
+  console.log(request.interceptResolution()); // ['already-handled'], meaning a legacy resolution has taken place
+  request.continue({}, 0); // Legacy behavior: throws an exception becasue abort() was already called
+});
+```
+
+In this example, Legacy Mode prevails and the request is continued because at least one handler does not specify a `priority`:
+
+```ts
+// Final outcome: immediate continue()
+page.setRequestInterception(true);
+page.on('request', (request) => {
+  request.abort('failed', 0); // Cooperative behavior: would execute at priority 0, but instead will throw an exception because legacy continue() gets called
+});
+page.on('request', (request) => {
+  console.log(request.interceptResolution()); // ['abort', 0], meaning an abort @ 0 is the current winning resolution
+  request.continue({}); // Legacy behavior: executes immediately, causing the previous cooperative abort() to throw an exception
+});
+```
+
+In this example, Cooperative Mode is active because all handlers specify a `priority`. `continue()` wins because it has a higher priority than `abort()`.
+
+```ts
+// Final outcome: cooperative continue()
+page.setRequestInterception(true);
+page.on('request', (request) => {
+  request.abort('failed', 0); // Cooperative behavior: aborts at priority 10
+});
+page.on('request', (request) => {
+  request.continue(request.continueRequestOverrides(), 5); // Cooperative behavior: continues at priority 5
+});
+page.on('request', (request) => {
+  console.log(request.interceptResolution()); // ['continue', 5], because continue @ 5 > abort @ 0
+});
+```
+
+In this example, Cooperative Mode is active because all handlers specify `priority`. `respond()` wins because its priority ties with `continue()`, but `respond()` beats `continue()`.
+
+```ts
+// Final outcome: cooperative respond()
+page.setRequestInterception(true);
+page.on('request', (request) => {
+  request.abort('failed', 10); // Cooperative behavior: aborts at priority 10
+});
+page.on('request', (request) => {
+  request.continue(request.continueRequestOverrides(), 15); // Cooperative behavior: continues at priority 15
+});
+page.on('request', (request) => {
+  request.respond(request.responseForRequest(), 15); // Cooperative behavior: responds at priority 15
+});
+page.on('request', (request) => {
+  console.log(request.interceptResolution()); // ['respond', 15], because respond @ 15 > continue @ 15 > abort @ 10
+});
+```
+
+##### Upgrading to Cooperative Mode for Package Maintainers
```

Comment:
We use sentence case for titles:

```suggestion
##### Upgrading to Cooperative Mode for package maintainers
```

("Cooperative Mode" is consistently capitalized even in sentences)